modules/core/keeper/keeper.go

```diff
@@ -50,12 +50,28 @@ func NewKeeper(
 	}
 
 	// panic if any of the keepers passed in is empty
-	if reflect.ValueOf(stakingKeeper).IsZero() {
```

Comment:
I think [`ValueOf`](https://pkg.go.dev/reflect#ValueOf) returns the zero value for `nil`, i.e. blank pointers.
Perhaps we could add some test cases before adapting this code to use the switch case.

Reply:
Hey @damiannolan! 

In the code from the [issue](https://github.com/cosmos/ibc-go/issues/2268) raised, the behavior of a pointer value was not as predicted with `ValueOf`

Will try to add additional test cases to confirm the need for switch case.

Another approach can be to add another `or` statement in the same line, but I thought switch case would have more readability plus a different error message for value & pointers.

Reply:
Could we modify the if statement to be:
```go
if stakingKeeper == nil || reflect.ValueOf(stakingKeeper).IsZero() {
```

I don't really want to over complicate this code. It is just a defensive check to prevent the chain from starting if the app.go is misconfigured. We simply want to know if the arguments provided are empty. I feel like there should be simple way of doing this? Otherwise we should add a private function to perform the switch so it just reads:

```go
assertNotEmpty(stakingKeeper)
assertNotEmpty(upgradeKeeper)
```

Reply:
I understand the concern but running the following code

```fmt.Println(stakingKeeper == nil)```

when given a pointer, returns `false`

I tried to solve it in a simpler way but failed. I can make a separate function in the same file that would check if an interface object(struct or pointer) is empty if you think that approach is better here

Reply:
I see the nil check won't catch the &stakingtypes.StakingKeeper{} issue. 

I think my preference would be to move the switch to a private assertNotEmpty function which returns an error which we panic on wrapping the name of the keeper:
```go
if isEmpty(stakingKeeper) {
    panic(fmt.Errorf("cannot initialize IBC keeper: empty staking keeper"))
}
```

Reply:
As @crodriguezvega mentioned, it's good to know if the error was due to value or pointer, if we make a function as you suggest, we won't be able to know the difference

Is that okay or do you suggest we think of something else?